### DISCUSSION TITLE
submission tcp/587 not listening

### DISCUSSION THREAD

**Discussion by tseatah** [Mailu/Mailu#3335] [User support] [answered]
Hi there, 

Trying to move over to Mailu, and I've run into a problem where the submission (tcp/587) port does not appear to have anything listening to it. Other ports that make use of certificates are working fine.

```
my.mail.fqdn [a.b.c.d] 25 (smtp) open
my.mail.fqdn [a.b.c.d] 80 (http) open
my.mail.fqdn [a.b.c.d] 110 (pop3) : Connection refused
my.mail.fqdn [a.b.c.d] 143 (imap2) : Connection refused
my.mail.fqdn [a.b.c.d] 443 (https) open
my.mail.fqdn [a.b.c.d] 465 (submissions) open
my.mail.fqdn [a.b.c.d] 587 (submission) : Connection refused
my.mail.fqdn [a.b.c.d] 993 (imaps) open
my.mail.fqdn [a.b.c.d] 995 (pop3s) open
```

Though _docker ps_ shows the _front_ container having the redirection to 587/tcp, there's nothing within that container that's listening:
```
root@corone:/data/mailu# docker compose exec front /bin/bash
fae3449d80aa:/app# netstat -tulpan | grep LISTEN
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      19/nginx -g daemon
tcp        0      0 0.0.0.0:4190            0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 0.0.0.0:25              0.0.0.0:*               LISTEN      19/nginx -g daemon
tcp        0      0 0.0.0.0:465             0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 0.0.0.0:2525            0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      19/nginx -g daemon
tcp        0      0 127.0.0.11:44097        0.0.0.0:*               LISTEN      -
tcp        0      0 0.0.0.0:995             0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 0.0.0.0:993             0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 0.0.0.0:14190           0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 127.0.0.1:8000          0.0.0.0:*               LISTEN      19/nginx -g daemon
tcp        0      0 0.0.0.0:10025           0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 127.0.0.1:10204         0.0.0.0:*               LISTEN      19/nginx -g daemon
tcp        0      0 0.0.0.0:10143           0.0.0.0:*               LISTEN      16/dovecot
tcp        0      0 :::80                   :::*                    LISTEN      19/nginx -g daemon
tcp        0      0 :::4190                 :::*                    LISTEN      16/dovecot
tcp        0      0 :::25                   :::*                    LISTEN      19/nginx -g daemon
tcp        0      0 :::465                  :::*                    LISTEN      16/dovecot
tcp        0      0 :::2525                 :::*                    LISTEN      16/dovecot
tcp        0      0 :::443                  :::*                    LISTEN      19/nginx -g daemon
tcp        0      0 :::995                  :::*                    LISTEN      16/dovecot
tcp        0      0 :::993                  :::*                    LISTEN      16/dovecot
tcp        0      0 :::14190                :::*                    LISTEN      16/dovecot
tcp        0      0 :::10025                :::*                    LISTEN      16/dovecot
tcp        0      0 :::10143                :::*                    LISTEN      16/dovecot
```

Looking at /etc/dovecot/dovecot.conf, it looks like the listener for the submission port is set to 0:
```
service submission-login {
   inet_listener submission {
    port = 0
   }
   inet_listener submissions {
    port = 465
    ssl = yes
   }

   inet_listener submission-webmail {
    port = 10025
   }
}
```

Why would that be? What is the proper way to have this listen to port 587?

**Comment by nextgens** [comment, marked as answer]
It's explained in the release notes.

**Reply by joni1802** [reply]
Where is it explained in the release notes? I can't find anything there (https://mailu.io/2.0/releases.html#mailu-2-0-2023-04-03). Can you please explain the reason behind that.

**Reply by nextgens** [reply]
The current stable release is 2024.06.

https://mailu.io/2024.06/releases.html#starttls-ports-disabled-by-default

**Reply by joni1802** [reply]
The blog post about "No STARTTLS" is interesting. But if I have understand everything correctly, most mail clients do not have the described vulnerability anymore and got fixed. The funny thing is that Port 465 got reassigned by the IANA and you should use 587 for authenticated SMTP.

**Reply by nextgens** [reply]
You have not understood properly.

SMTP is for mailserver to mailserver comms. SMTPS is dead (as in will never happen).

Port 465 is assigned to submissions (what you call authenticated SMTP which is in fact a different protocol)... and that is very much alive and well. You should use that instead of port 587 (which is submission).

**Reply by joni1802** [reply]
Oh I am seeing that since 2018 (RFC 8314) it is recommended to use implicit TLS over port 465. Most articles on the internet still referencing port 465 with SMTPS and saying it is deprecated. That's confusing as fuck. Thank you, I have learnt something new. But wouldn't it be nice to still optionally support STARTTLS over port 587 in Mailu?

**Reply by nextgens** [reply]
We do, you just have to set the PORTS variable, as described in the release notes.

**Comment by tseatah** [comment]
Of course... I _knew_ I had seen it somewhere :P
